pkg/analysis/passes/legacyplatform/legacyplatform.go

```diff
@@ -27,11 +27,11 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	)
 
 	if angularExp.Match(module) && !reactExp.Match(module) {
-		pass.Reportf(pass.AnalyzerName, legacyPlatform, "module.js: uses legacy plugin platform")
+		pass.ReportResult(pass.AnalyzerName, legacyPlatform, "module.js: uses legacy plugin platform", "The plugin uses the legacy plugin platform (angularjs). Please migrate the plugin to use the new plugins platform.")
 	} else {
 		if legacyPlatform.ReportAll {
 			legacyPlatform.Severity = analysis.OK
-			pass.Reportf(pass.AnalyzerName, legacyPlatform, "module.js: uses current plugin platform")
+			pass.ReportResult(pass.AnalyzerName, legacyPlatform, "module.js: uses current plugin platform", "")
```

Comment:
(Not related to the new messages) I see this pattern repeated a lot across the various passes we have:

```go
x.Severity = analysis.OK
pass.ReportResult(pass.AnalyzerName, x, "some message", "")
```

(setting severity to OK and empty string as details).

I feel like we can combine those in a single function call? Maybe in another PR? I'll see if we can come up with something

Reply:
Indeed. We could have "ReportIssue" and "ReportOK" functions.

Reply:
:+1: Agree